### ISSUE TITLE
`predict` bug with mismatched data types

### ISSUE THREAD

**Issue by ivanhigueram** [py-econometrics/pyfixest#731]
Hello there, 

I am trying to create predictions with a dataset outside my model data. I found that if there's any type mismatch in the `newdata` compared with the `data` we used to estimate the model, the `predict()` method will return an array of `nan`: 

Here's a reproducible example:

```python
import pyfixest as pf

data = pf.get_data()

# Run model
model = pf.feols("Y ~ X1 | f1", data=data)

# Allow me to fill the nans just to make my points :)
data["f1"] = data["f1"].fillna(0).astype(int)
model.predict(newdata=data.iloc[0:100])

>>> array([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, ...])
```

This error is coming from the definition of `df_fe` in the `predict()` and the `_apply_fixef_numpy` as the dictionary keys will be saved with a `10.0` rather than `10`. 
https://github.com/py-econometrics/pyfixest/blob/10a8fb8417e5c7581bf6ae5824f1aa6221f1d261/pyfixest/estimation/feols_.py#L1832

Not sure if this constitutes a bug on itself, or if this is a skill issue, but it would be nice to get a warning maybe? This is no problem for unit FEs if the ID is a `str`, but in dates, sometimes we get `2010.0` rather than `2010` when we do time operations and numpy preserves the data type. 

I am running the `'0.26.2'` version in Python `3.10`.

**Comment by s3alfisc** [comment]
Thanks for reporting this Ivan @ivanhigueram! I have to think about this one for a bit. My intuition would be that a keys of 10.0 and 10 should be treated equally? For sure I think adding a warning would be a good start!

**Comment by s3alfisc** [comment]
If you had to choose, what would be your preferred behavior?

**Comment by ivanhigueram** [comment]
I'd say a warning would be enough. Is definitely easier to do a `astype()` in the `newdata` than trying to fix this in your code base.

**Comment by s3alfisc** [comment]
Yes I was also afraid that handling things in pyfixest would be a lot of work 😅 I'll add a warning then (or would you be up to open a PR? but of course no pressure 😄 ). 

I think it could be as simple as adding 

```python
        if self._has_fixef: 
            fixef = self._fixef.split("+")

            mismatched_fixef_types = [x for x in fixef if newdata[x].dtypes != self._data[x].dtypes]
            if mismatched_fixef_types:
                warnings.warn(f"Data types of fixed effects {mismatched_fixef_types} do not match the model data. This leads to     mismatched keys in the fixed effect dictionary, and as a result, to NaN predictions for columns with mismatched keys.")        
```
around here https://github.com/py-econometrics/pyfixest/blob/611e5ccdfd2a52cd41890e328bb5358f7ed240c9/pyfixest/estimation/feols_.py#L1811 

and to add a test that triggers the warning in [test_errors.py](https://github.com/py-econometrics/pyfixest/blob/master/tests/test_errors.py).

**Comment by ivanhigueram** [comment]
I'd love to open the PR. I will ping if I go into any problems with testing and all that.

**Comment by ivanhigueram** [comment]
@s3alfisc let me know if you need any comments. I didn't run tests locally because it seems a small change, but let me know if I should (should I `pixi run test`? -- I am loving this `pixi).

**Comment by s3alfisc** [comment]
> I am loving this `pixi

Big fan as well! 

No need to run tests locally, we can let the CI handle it =)

**Comment by s3alfisc** [comment]
@all-contributors please add @ivanhigueram for code
